### PR TITLE
add language DirectX 3D File

### PR DESCRIPTION
**Pull request by malytomas**
## Description

See https://github.com/github/linguist/issues/4679

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?l=&q=extension%3Ax+size%3A%3E10000+%22xof+0303txt+0032%22&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
    - Sample license(s):
       The file added to samples is model of a cube, created and exported from blender. CC0.
  
  - [ ] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=cson&grammar_url=https%3A%2F%2Fgithub.com%2FAlhadis%2Flanguage-etc%2Fblob%2Fmaster%2Fgrammars%2Fdirect-x.cson&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fdinghim%2Fmagic_cube%2Fblob%2Fd4aa1b0d3438ce39c3d90c6a77a161fe69cfe0b8%2FmagicCube%2Fcube.X&code=
     - the command scripts/add-grammar is failing: cannot connect to docker daemon.
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.
    - help needed :D

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -1132,6 +1132,11 @@ Diff:
   codemirror_mode: diff
   codemirror_mime_type: text/x-diff
   language_id: 88
+DirectX 3D File:
+  type: data
+  extensions:
+  - ".x"
+  language_id: 201049282
```
Comment (Alhadis) [inline lib/linguist/languages.yml]: Alright, first, let's add the syntax highlighting fields:

~~~yaml
- ace_mode: text
- tm_scope: source.direct-x
~~~

(The [ACE editpr](https://ace.c9.io) doesn't have that many editing modes, so we usually leave this field as `text`. It's a mandatory field, however).

**Comment by Alhadis** [comment]
> the command `scripts/add-grammar` is failing: cannot connect to docker daemon

In this instance, you actually don't need to run that script; I've already added the grammar to a repository that's already added as a submodule. What you *do* need to do is update it manually; I'll handle this by pushing changes to your branch, since it's a little complicated.

File: lib/linguist/languages.yml (hunk 1 above)
Comment (malytomas) [reply lib/linguist/languages.yml]: what about the codemirror_* fields?

Comment (Alhadis) [reply lib/linguist/languages.yml]: Nope, those are optional. CodeMirror [doesn't have that many language-modes either](https://codemirror.net/mode/index.html), so unless it has one that produces similar-enough highlighting, we generally just leave these fields out (it defaults to `text/plain` in any case).

**Comment by Alhadis** [comment]
@malytomas I can't push to your branch. Could you check that you have this checkbox ticked on the right-hand side of this page, please? :)

Figure 1

Koszi!

**Comment by malytomas** [comment]
> 
> 
> @malytomas I can't push to your branch. Could you check that you have this checkbox ticked on the right-hand side of this page, please? :)
> 
> Figure 1
> 
> Koszi!

It is. I have also added you as a collaborator.

**Comment by Alhadis** [comment]
Okay, that shouldn't be happening. GitHub also indicated I could push to your branch, but my push was rejected:

Figure 2

So, uh, apologies for making this PR more complicated than it had to be. 😁

**Comment by Alhadis** [comment]
BTW, when forking for a pull-request, it always helps to create a separate branch for your changes (instead of working off your forked `master` branch). It makes stuff like this a bit messier to deal with.

**Comment by malytomas** [comment]
> BTW, when forking for a pull-request, it always helps to create a separate branch for your changes (instead of working off your forked `master` branch). It makes stuff like this a bit messier to deal with.

Apologies. I never had issues like this before.
Anyway, glad you figured it out. :D

Is there anything else I can help with? :D
(Not that I helped much so far... :( )

**Comment by Alhadis** [comment]
Don't worry, this is my fault. I should have rolled the grammar into [my other pull-request](https://github.com/github/linguist/pull/), which is the reason for all our troubles (I think)...

To make things worse, I mistakenly merged the wrong file locally, causing the [license-hash](https://github.com/github/linguist/commit/a1925c150d81de7a16188247e6003b2cf1f681ef/checks?check_suite_id=272805228) to change erroneously... 😥 (This isn't a normal part of the procedure for adding grammars to Linguist, so don't let this dissuade you from future contributions. 😀)

**Comment by Alhadis** [comment]
Ugh, the frigging tests are passing locally, I've no idea what's going on with the license cache. 😞

Tell you what, let's try again, but *without* the grammar this time. Just put `none` as the `tm_scope` field's value; I can add it in a follow-up PR if need be (probably in #4678).

Once again, sorry for this. 😢

**Pull request closed without merging**